addons/website/static/src/js/menu/new_content.js

```diff
@@ -65,23 +103,187 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
+                    if (this.firstTab || document.activeElement === this.$firstLink[0]) {
+                        this._focusLastLink();
+                        ev.preventDefault();
+                    }
+                } else {
+                    if (this.firstTab || document.activeElement === this.$lastLink[0]) {
+                        this._focusFirstLink();
+                        ev.preventDefault();
+                    }
+                }
+                this.firstTab = false;
+                break;
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Set the focus on the first link
+     *
+     * @private
+     */
+    _focusFirstLink: function () {
+        this.$firstLink.focus();
+    },
+    /**
+     * Set the focus on the last link
+     *
+     * @private
+     */
+    _focusLastLink: function () {
+        this.$lastLink.focus();
+    },
+    /**
+     * Hide the menu
+     *
+     * @private
+     */
+    _hideMenu: function () {
         this.$newContentMenuChoices.addClass('o_hidden');
+        $('body').removeClass('modal-open');
+    },
+    /**
+     * Install a module
+     *
+     * @private
+     * @param {number} moduleId: the module to install
+     */
+    _install: function (moduleId) {
+        this.pendingInstall = 1;
+        return this._rpc({
+            route: '/website/module/' + moduleId + '/install',
+        });
+    },
+    /**
+     * Open the install dialog related to an element:
+     *  - open the dialog depending on access right and another pending install
+     *  - if ok to install, prepare the install action:
+     *      - call the proper action on click
+     *      - change the button text and style
+     *      - handle the result (reload on the same page or error)
+     *
+     * @private
+     * @param {object} $el: the related element
+     */
+    _openInstallDialog: function ($el) {
+        var self = this;
+        var $i = $el.find('a i');
+        var $p = $el.find('a p');
+
+        var title = $p.text();
+        var content = '';
+        var buttons;
+
+        var moduleId = $el.data('module-id');
+
+        var name = $el.data('module-shortdesc');
+        var canInstall = $el.data('module-can-install');
+
+        function rpcDone(result) {
+            if (!result) {
+                rpcFail();
+                return;
+            }
+            window.location.href = cleanUrl(window.location.href, true, true) + '?' + enableFlag;
+        }
+        function rpcFail() {
+            $i.removeClass()
+                .addClass('fa fa-exclamation-triangle');
+            $p.text(_.str.sprintf(_t('"%s" failed to install.'), name));
+        }
+
+        if (this.pendingInstall) {
+                content = _.str.sprintf(_t('The installation of a module is already in progress.'), name);
```

Comment:
I don't think we'll be able to translate the text while injecting the module name if we make it in xml? (not this one because it doesn't have a %s but the others?)